libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2412,6 +2412,29 @@ string YulUtilFunctions::conversionFunctionSpecial(Type const& _from, Type const
 			.render();
 		}
 
+		if (_from.category() == Type::Category::Array && _to.category() == Type::Category::Array)
+		{
+			auto const& fromArrayType = dynamic_cast<ArrayType const&>(_from);
+			auto const& toArrayType = dynamic_cast<ArrayType const&>(_to);
+
+			solAssert(!fromArrayType.baseType()->isDynamicallyEncoded(), "");
+			solAssert(fromArrayType.isByteArray() && toArrayType.isByteArray(), "");
```

Comment:
Shouldn't this be an unimplementedassert?

Reply:
(and all other assertions that follow, too?)

Reply:
Indeed, changed now